### ISSUE TITLE
Verbose GC logs do not work on OSX

### ISSUE THREAD

**Issue by rwy0717** [eclipse/openj9#9015]
When I build and run j9 on macOS (slowdebug, with **fno-inline**), I get a warning about undefined symbols:

```
JVMJ9VM011W Unable to load j9vrb29: dlopen(/Users/rwyoung/wsp/openj9-openjdk-jdk/build/macosx-x86_64-server-fastdebug/jdk/lib/default/libj9vrb29.dylib, 2): Symbol not found: _get16BitByteCodeIndexFromExceptionTableVerbose
   Referenced from: /Users/rwyoung/wsp/openj9-openjdk-jdk/build/macosx-x86_64-server-fastdebug/jdk/lib/default/libj9vrb29.dylib
   Expected in: flat namespace  in /Users/rwyoung/wsp/openj9-openjdk-jdk/build/macosx-x86_64-server-fastdebug/jdk/lib/default/libj9vrb29.dylib
```

Failing to load this library effectively disables all verbose logging. This appears to be a long standing issue, first mentioned [here](https://github.com/eclipse/openj9/issues/36#issuecomment-420837674), discovered during our initial port to OSX.

**Comment by rwy0717** [comment]
`MethodMetaData.h` declares a number of `__inline` functions, which are in turn defined in `MethodMetaData.c`, which is a source for the `j9vrb.dylib` library. Since this is a C source, we get C99 inline semantics. In C99, If the compiler chooses to not inline a function (or can't, because of `-fno-inline`), then it will expect to link against a different definition (which will be non-inline or `extern`). We don't define any such thing.

I believe this error does not appear with GCC, because we also use `__attribute__((force_inline))`.

Since these functions are only used in `MethodMetaData.c`, they should be made into private, `static inline` functions.

**Comment by rwy0717** [comment]
fix: #9016
